nixos/modules/services/networking/headscale.nix

```
@@ -464,19 +539,17 @@ in {
 
     systemd.services.headscale = {
       description = "headscale coordination server for Tailscale";
-      wants = [ "network-online.target" ];
+      wants = ["network-online.target"];
       after = ["network-online.target"];
       wantedBy = ["multi-user.target"];
       restartTriggers = [configFile];
 
-      environment.GIN_MODE = "release";
-
       script = ''
-        ${optionalString (cfg.settings.db_password_file != null) ''
-          export HEADSCALE_DB_PASS="$(head -n1 ${escapeShellArg cfg.settings.db_password_file})"
+        ${optionalString (cfg.settings.database.postgres.password_file != null) ''
+          export HEADSCALE_DATABASE_POSTGRES_PASS="$(head -n1 ${escapeShellArg cfg.settings.database.postgres.password_file})"
         ''}
 
-        exec ${cfg.package}/bin/headscale serve
+        exec ${cfg.package}/bin/headscale serve --config ${configFile}
```

The goal of this change is to make restarting of the service when settings have changed more reliable, I found that using the etc path was not always doing it.

The splitting of the CLI and server config also allows for the configuration to be ran without getting all the errors of the server config whenever the user is running the CLI.

---

```suggestion
        exec ${lib.getExe cfg.package} serve --config ${configFile}
```



> The splitting of the CLI and server config also allows for the configuration to be ran without getting all the errors of the server config whenever the user is running the CLI.

Otherwise I personally would have never noticed half of them.

Also this could easily be achieved by using restartTriggers.

I always prefer using /etc/ because that allows to quickly change the file in place to quickly debug things.